### PR TITLE
Add copy visual feedback to tx share button

### LINKED ISSUE #3283: No visual feedback when clicking on the Tx Share button
## Bug description

When clicking on the Share button in transaction details, there's no tooltip saying "Copied".

Screenshot 2024-02-19 at 15 00 53

### PR DESCRIPTION
**Pull request by daniel0ar**
## What it solves

Resolves #3283

## How this PR fixes it
It uses the existing CopyTooltip common compoent inside the TxShareLink component.

## How to test it
Open http://localhost:3000/transactions/tx?safe=oeth:0x773D7c58941295aF412092efF5Aa3d188525e81a&id=multisig_0x773D7c58941295aF412092efF5Aa3d188525e81a_0x96afb4ae2fc570f3c52f53f820556a430b23ffc610f108d9e549fcf64c99ee3f and click on the Share button 

## Screenshots
image

## Checklist
* [x] Wrapped the Share IconButton with the CopyTooltip component
* [x] Passed the url to the tooltip component as text prop
* [x] Removed onCopy action (the tooltip component copies the text passed) 

Note: At first I added a new ExportTooltip component, then realized it was not necesarry unless the share button changes its action functionality (i.e: doing something other than copying tx url), so reverted first commit and used existing Copytooltip component

### CODE REVIEW

**Comment by daniel0ar** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/components/transactions/TxShareLink/index.tsx [hunk 1]
```diff
@@ -1,45 +1,26 @@
...
-
-    try {
-      // copy href to clipboard
-      navigator.clipboard.writeText(location.origin + href)
-    } catch (error) {
-      console.error(error)
-      setIsCopyEnabled(false)
-    }
-  }
+  const tx_url = location.origin + href
```
[the remark this answers is not in the document]
Comment (daniel0ar) [reply src/components/transactions/TxShareLink/index.tsx:15]: Oh ok. Should I wrap it inside a useEffect then? like:
```ts
let txUrl = ""
useEffect(() => {
    txUrl = location.origin + href
}, [href])
```

or pass it directly as prop like:
```ts
<CopyTooltip text={location.origin + href}>
```

[the remark this answers is not in the document]
Comment (katspaugh) [reply src/components/transactions/TxShareLink/index.tsx:15]: Neither is a good practice. In the first case `txUrl` is an implicit dependency of the hook, and the second one is no different from the original version.


I would do it like this:

```js
// Outside of the component
const useOrigin = () => {
  const [origin, setOrigin] = useState('')

  useEffect(() => {
    if (typeof location !== 'undefined') {
      setOrigin(location.origin)
    }
  }, [])
  
  return origin
}


// In the component 
const txUrl = useOrigin() + href
```

File: src/components/transactions/TxShareLink/index.tsx [hunk 2]
```diff
@@ -1,45 +1,26 @@
...
-        data-testid="share-btn"
-        component={Link}
-        aria-label="Share"
-        href={href}
-        onClick={onClick}
-        disabled={!isCopyEnabled}
-      >
-        <SvgIcon component={ShareIcon} inheritViewBox fontSize="small" color="border" />
-      </IconButton>
+      <CopyTooltip text={tx_url}>
```
[the remark this answers is not in the document]
Comment (daniel0ar) [reply src/components/transactions/TxShareLink/index.tsx]: Done!

File: src/components/transactions/TxShareLink/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (daniel0ar) [reply src/components/transactions/TxShareLink/index.tsx]: Done! Thanks for the feedback and info.

**Review by katspaugh — APPROVED** [review]
Awesome, thank you!

**Pull request merged**